File: nattybear/level2/짝지어-제거하기/solution.js
function solution(s) {
  for (let i = 0; i < s.length; i++)
    push(s[i])
  if (empty())
    return 1
  else
    return 0
}

const stack = []

function push(x) {
  if (empty())
    stack.push(x)
  else {
    if (x == peek())
      stack.pop()
    else
      stack.push(x)
  }
}

function empty() {
  return stack.length == 0
}

function peek() {
  const size = stack.length
  return stack[size - 1]
}

File: nattybear/level2/짝지어-제거하기/Solution.java
import java.util.ArrayList;
import java.util.Stack;

class Solution {
  public int solution(String s) {
    CStack stack = new CStack();
    for (int i = 0; i < s.length(); i++)
      stack.foo(s.charAt(i));
    if (stack.empty())
      return 1;
    else
      return 0;
  }
}

class CStack extends Stack<Character> {
  void foo(Character c) {
    if (empty())
      push(c);
    else {
      if (peek() == c)
        pop();
      else
        push(c);
    }
  }
}
